chemclipse/plugins/org.eclipse.chemclipse.swt.ui/src/org/eclipse/chemclipse/swt/ui/components/SearchSupportUI.java

```diff
@@ -152,18 +152,21 @@ public void widgetDefaultSelected(SelectionEvent e) {
 		return text;
 	}
 
-	private Button createCheckBoxCaseSensitive(Composite parent) {
+	private Button createButtonCaseSensitive(Composite parent) {
 
-		Button button = new Button(parent, SWT.CHECK);
+		Button button = new Button(parent, SWT.PUSH);
 		button.setText("");
 		button.setToolTipText("Search Case Sensitive");
-		button.setSelection(PreferenceSupplier.isSearchCaseSensitive());
+		boolean active = PreferenceSupplier.isSearchCaseSensitive();
```

Comment:
@Mailaender The boolean variable `active` is only used once here. Probably move the `PreferenceSupplier.isSearchCaseSensitive()` into the set image method:

`button.setImage(ApplicationImageFactory.getInstance().getImage(IApplicationImage.IMAGE_CASE_SENSITIVE, IApplicationImage.SIZE_16x16, PreferenceSupplier.isSearchCaseSensitive()));`

Reply:
I was worrying about the line length, but you are right.